src/components/Filter/Filter.tsx

```diff
@@ -0,0 +1,56 @@
+import classNames from 'classnames';
+import { useState } from 'react';
+
+type Props = {
+  setFilter: (filter?: boolean) => void
+};
+
+enum TypeFilter {
+  ALL = 'all',
+  ACTIVE = 'active',
+  COMPLETED = 'completed',
+}
+
+export const Filter: React.FC<Props> = ({ setFilter }) => {
+  const [typeFilter, setTypeFilter] = useState(TypeFilter.ALL);
+
+  const chooseFilter = (type: TypeFilter, filter?: boolean) => {
+    setTypeFilter(type);
+    setFilter(filter);
+  };
+
+  return (
+    <nav className="filter">
+      <a
+        href="#/"
+        className={classNames('filter__link', {
+          selected: typeFilter === TypeFilter.ALL,
+        })}
+        onClick={() => chooseFilter(TypeFilter.ALL)}
+      >
+        All
+      </a>
+
+      <a
+        href="#/active"
+        className={classNames('filter__link', {
+          selected: typeFilter === TypeFilter.ACTIVE,
+        })}
+        onClick={() => chooseFilter(TypeFilter.ACTIVE, false)}
+      >
+        Active
+      </a>
+
+      <a
+        href="#/completed"
+        className={classNames('filter__link', {
+          selected: typeFilter === TypeFilter.COMPLETED,
+        })}
+        onClick={() => chooseFilter(TypeFilter.COMPLETED, true)}
+      >
+        Completed
+      </a>
+    </nav>
+
+  );
+};
```

Comment:
Also, will be better if you put this to the separate component
```suggestion
  };
  
  const filters = [TypeFilter.ALL, TypeFilter.ACTIVE, TypeFilter.COMPLETED];

  return (
    <nav className="filter">
    {filters.map((type) => (
      <a
        href="#/"
        className={classNames('filter__link', {
          selected: typeFilter === type,
        })}
        onClick={() => chooseFilter(type)}
      >
        {`${type.slice(0, 1).toUpperCase()}${tyle.slice(1)}`}
      </a>
      )}
    </nav>

  );
};
```